projects/valtimo/access-control-management/src/lib/components/editor/access-control-editor.component.ts

```diff
@@ -0,0 +1,205 @@
+          this.showSuccessMessage(this._roleKey);
+          this.setModel(result);
+        },
+        error: () => {
+          this.enableMore();
+          this.enableSave();
+          this.enableEditor();
+        },
+      });
+  }
+
+  public onDelete(roles: Array<string>): void {
+    this.disableEditor();
+    this.disableSave();
+    this.disableMore();
+
+    this.accessControlService.dispatchAction(
+      this.accessControlService.deleteRoles({roles}).pipe(
+        finalize(() => {
+          this.router.navigate(['/access-control']);
+        })
+      )
+    );
+  }
+
+  public showDeleteModal(): void {
+    this.showDeleteModal$.next(true);
+  }
+
+  public showEditModal(): void {
+    this.showEditModal$.next(true);
+  }
+
+  public onEdit(currentRoleKey: string, data: Role | null): void {
+    this.showEditModal$.next(false);
+
+    if (!data) {
+      return;
+    }
+
+    this.disableEditor();
+    this.disableSave();
+    this.disableMore();
+
+    this.accessControlService.updateRole(currentRoleKey, data).subscribe(() => {
+      this.router.navigate([`/access-control/${data.roleKey}`]);
+      this.showSuccessMessage(data.roleKey);
+    });
+  }
+
+  private openRoleKeySubscription(): void {
+    this._roleKeySubscription = this.route.params
+      .pipe(
+        filter(params => params?.id),
+        map(params => params.id),
+        tap(roleKey => {
+          this._roleKey = roleKey;
+          this.pageTitleService.setCustomPageTitle(roleKey, true);
+          this.deleteRowKeys$.next([roleKey]);
+        }),
+        switchMap(roleKey => this.accessControlService.getRolePermissions(roleKey)),
+        tap(permissions => {
+          this.enableMore();
+          this.enableSave();
+          this.enableEditor();
+          this.setModel(permissions);
+        })
+      )
+      .subscribe();
+  }
+
+  private getPermissions(): void {
+    this.route.params
+      .pipe(
+        tap(params => {
+          this.pageTitleService.setCustomPageTitle(params?.id);
+          this.deleteRowKeys$.next([params?.id]);
+        }),
+        switchMap(params => this.accessControlService.getRolePermissions(params.id))
+      )
+      .subscribe(permissions => {
+        this.enableMore();
+        this.enableSave();
+        this.enableEditor();
+        this.setModel(permissions);
+      });
+  }
+
+  private setModel(permissions: object): void {
+    this.model$.next({
+      value: JSON.stringify(permissions),
+      language: 'json',
+    });
+  }
+
+  private disableMore(): void {
+    this.moreDisabled$.next(true);
+  }
+
+  private enableMore(): void {
+    this.moreDisabled$.next(false);
+  }
+
+  private disableSave(): void {
+    this.saveDisabled$.next(true);
+  }
+
+  private enableSave(): void {
+    this.saveDisabled$.next(false);
+  }
+
+  private disableEditor(): void {
+    this.editorDisabled$.next(true);
+  }
+
+  private enableEditor(): void {
+    this.editorDisabled$.next(false);
+  }
```

Comment:
Instead of 6 methods, do you think it would be a better idea to condense it down to 3 parametrized? Something like toggleMore(enable: boolean): void